### ISSUE TITLE
Single file usage of nix-buffer

### ISSUE THREAD

**Issue by matthewbauer** [shlevy/nix-buffer#8]
Right now, I don't think there is a way to get nix-buffer to work without a separate dir-locals.nix. It seems like there should be a way to set it up entirely through the `file local variables'.

For instance, I have a Haskell file that doesn't have any Stack/Cabal/etc. dependencies associate with it. Is there a way I can include dependencies in it?

I basically want to setup something like this:

```
-- Local Variables:
-- nix-buffer-script: "with import <nixpkgs> {}; nixBufferBuilders.withPackages [
--   haskellPackages.ghcWithPackages (pkgs: with pkgs; [ random QuickCheck parsec ])
-- ]"
-- eval: (and (require 'nix-buffer nil t) (nix-buffer))
-- End:
```

and have it work out of the box. Is that seem reasonable? This is just me brainstorming  how to do this right now.

**Comment by shlevy** [comment]
You could possibly use nix-buffer--nix-build for this already, but I wouldn't object to a better interface

**Comment by matthewbauer** [comment]
Do you think it's okay to bypass the safety checks? In this case there's not really a "file" in the same way as for dir-locals.nix. I guess I can make it just use the file name alternatively.

**Comment by shlevy** [comment]
Yeah, you can bypass the security checks, you're already running a file-local eval

**Comment by shlevy** [comment]
@matthewbauer Let me know if you plan to pick this up again, or we can close it out for now.
